### PR TITLE
feat: add customization compute pod node

### PR DESCRIPTION
**Pull request by guilhem-barthes**
Fixes FL-1674

### CODE REVIEW

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

File: docs/source/how-to/deploying-substra/howto/customize-compute-pod-node.rst [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+                        matchExpressions:
+                        - key: statefulset.kubernetes.io/pod-name
+                        operator: In
+                        values:
+                        - $(POD_NAME)
+                    topologyKey: kubernetes.io/hostname
+            ...
+
+
+We can see that set the value empty for ``nodeSelector`` and ``tolerations`` but defines an affinity rule that would force the pod to be spawned in the same pod than the worker.
```
Comment (SdgJlbl) [inline docs/source/how-to/deploying-substra/howto/customize-compute-pod-node.rst:47]:
```suggestion
We can see that it sets empty values for ``nodeSelector`` and ``tolerations`` but defines an affinity rule that would force the pod to be spawned in the same pod than the worker.
```

File: docs/source/how-to/deploying-substra/howto/customize-compute-pod-node.rst [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+.. note::
+    To allow flexibility in the way we define our ``nodeSelector``, ``affinity`` and ``tolerations``, we provide the following environment variables to use as `dependent environment variables <https://kubernetes.io/docs/tasks/inject-data-application/define-interdependent-environment-variables/>`_:
+    
+    - ``POD_NAME``: provide the name of the worker spawning the compute function
+    - ``NODE_NAME``: provide the name of the node on which the worker spawning the compute function is
+
+On-demand GPU
+=============
+
+In this section, we are using as an example use a dedicated node-pool to run the functions pods, provisioning nodes only when required. This example also feature using GPU sharing.
```
Comment (SdgJlbl) [inline docs/source/how-to/deploying-substra/howto/customize-compute-pod-node.rst:58]:
```suggestion
In this section, we are using as an example a dedicated node-pool to run the functions pods, provisioning nodes only when required. This example also features using GPU sharing.
```

File: docs/source/how-to/deploying-substra/howto/customize-compute-pod-node.rst [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+For the following example, we will assume that you have 2 node-pools:
+
+- ``node-pool-default``, the node-pool without GPU
+- ``node-pool-gpu``, the node-pool with gpu
+
+For our example, we assume ``node-pool-gpu`` has the following:
+
+- labels:
+    ``node-type=substra-tasks-gpu``
+    ``node_pool=node-pool-gpu``
```
Comment (SdgJlbl) [inline docs/source/how-to/deploying-substra/howto/customize-compute-pod-node.rst:72]:
```suggestion
    - ``node-type=substra-tasks-gpu``
    - ``node_pool=node-pool-gpu``
```

File: docs/source/how-to/deploying-substra/howto/customize-compute-pod-node.rst [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+                    max_shared_clients_per_gpu = <pod_count>
+                    ...
+                },
+                ...
+            ]
+        ...
+        }
+        
+
+In your value file, add the following:
```
Comment (SdgJlbl) [inline docs/source/how-to/deploying-substra/howto/customize-compute-pod-node.rst:179]:
```suggestion
In your values file, add the following:
```

**Review by SdgJlbl — APPROVED** [review]
Thanks a lot, great documentation, very clear and easy to follow, with the relevant links to go further on the topic 👍 
Some minor typos to fix before merge.

File: docs/source/how-to/deploying-substra/howto/customize-compute-pod-node.rst [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+            ...
+            nodeSelector:
+                ...
+                cloud.google.com/gke-gpu-sharing-strategy: time-sharing
+                cloud.google.com/gke-max-shared-clients-per-gpu: <pod_count>
+
+Other providers
+---------------
+
+For other providers, we recommend reading directly the documentation from your provider. If you're using a Nvidia GPU, you can read the reference on sharing GPU between pods (`Time-slicing <https://docs.nvidia.com/datacenter/cloud-native/gpu-operator/latest/gpu-sharing.html>`_ and `Multiple instance GPU (MIG) <https://docs.nvidia.com/datacenter/cloud-native/gpu-operator/latest/gpu-operator-mig.html>`_)
```
Comment (SdgJlbl) [inline docs/source/how-to/deploying-substra/howto/customize-compute-pod-node.rst:194]:
```suggestion
Other cloud providers
---------------

For other cloud providers, we recommend reading directly the documentation from your provider. If you're using a Nvidia GPU, you can read the reference on sharing GPU between pods (`Time-slicing <https://docs.nvidia.com/datacenter/cloud-native/gpu-operator/latest/gpu-sharing.html>`_ and `Multiple instance GPU (MIG) <https://docs.nvidia.com/datacenter/cloud-native/gpu-operator/latest/gpu-operator-mig.html>`_)
```

File: docs/source/how-to/deploying-substra/howto/customize-compute-pod-node.rst [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+
+Other graphic card providers
```
Comment (SdgJlbl) [inline docs/source/how-to/deploying-substra/howto/customize-compute-pod-node.rst:196]:
```suggestion
Other GPU providers
```

**Review by SdgJlbl — COMMENTED** [review]
And also, please add a changelog fragment, thanks :)

**Pull request merged**